idaes/apps/grid_integration/forecaster.py

```diff
@@ -681,3 +681,57 @@ def fetch_day_ahead_stats_from_prescient(self, uc_date, uc_hour, day_ahead_resul
                 self._historical_da_prices[b] = self._historical_da_prices[b][24:]
 
         return
+
+
+class PerfectForecaster(AbstractPrescientPriceForecaster):
+    def __init__(self, data_path_or_df):
+        """
+        Perfect forecaster that reads the data from a Dataframe containing:
+         - {bus}-DALMP
+         - {bus}-RTLMP
+         - {bus}-DACF and {bus}-RTCF for renewable plants
+        """
```

Comment:
Perhaps add a sentence explaining the perfect forecast has no error and "sees into the future".

Reply:
@Xinhe-Chen, did you do this? If so, please reply here.

Reply:
Done